packages/spacecat-shared-rum-api-client/src/functions/form-vitals.js

```diff
@@ -135,6 +137,42 @@ function findFormCTAForInternalNavigation(bundles, formVitals) {
   });
 }
 
+function findFormCTAWithinPage(bundles, formVitals) {
+  return formVitals.map((item) => {
+    const { url, formsource } = item;
+    const dataChunks = new DataChunks();
+    dataChunks.load([{ rumBundles: bundles }]);
+    dataChunks.addFacet('viewblock.source', checkpointSource('viewblock'));
+    dataChunks.addFacet('checkpoint', checkpointFacet);
+    dataChunks.addFacet('url', urlFacet);
+    dataChunks.filter = { checkpoint: ['click', 'viewblock'], url: [url], 'viewblock.source': [formsource] };
+    const sortedBundles = dataChunks.filtered.sort((a, b) => a.timeDelta - b.timeDelta);
+    const sources = sortedBundles.map((a) => {
+      const { events } = a;
+      const viewblock = events.find(({ checkpoint, source }) => checkpoint === 'viewblock' && source === formsource);
+      const viewblockTime = viewblock.timeDelta;
+
+      const clicks = events.filter((e) => e.checkpoint === 'click')
+        // clicks within 1 second of viewblock
+        .filter((e) => Math.abs(e.timeDelta - viewblockTime) < 1000)
+        // ignore form clicks
+        .filter((e) => e.source && !e.source.match(/\bform\b/i));
+
+      return clicks.map((_) => _.source);
+    }).filter((_) => _.length > 0).flat();
+    if (sources.length > 0) {
+      return {
+        ...item,
+        cta: {
+          sources,
+          form: item.url,
```

Comment:
form not required since this is same as the top url,
also need to make sources unique